### PR TITLE
ITSASU-1312, updated deleteBusniess in RemoveBusinessController

### PR DESCRIPTION
**Pull request by WozCapGem**

### CODE REVIEW

File: test/services/RemoveBusinessServiceSpec.scala [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      "a reference, business id and single selfEmploymentsData are passed into the service" in {
+        mockSaveSelfEmployments[Seq[SelfEmploymentData]](BusinessesKey, Seq.empty)(Right(PostSubscriptionDetailsSuccessResponse))
+        mockDeleteSubscriptionDetails(BusinessAccountingMethod)(Right(DeleteSubscriptionDetailsSuccessResponse))
+        val result = await(
+
+        TestRemoveBusiness.deleteBusiness(testReference, testBusinessId, Seq(testBusiness("id")))
+        )
+
+        result shouldBe Unit
+
```
Comment (sb-dev) [inline test/services/RemoveBusinessServiceSpec.scala:58]: Should we verify that all the mocks have been called?

Comment (AlexRimmerHMRC) [reply test/services/RemoveBusinessServiceSpec.scala:58]: The test fails without the mocks as it throws an exception, thus proving the mock was called.
Happy to discuss further

File: test/services/RemoveBusinessServiceSpec.scala [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      "a reference, business id and multiple selfEmploymentsData are passed into the service" in {
+        mockSaveSelfEmployments[Seq[SelfEmploymentData]](BusinessesKey, Seq(testBusiness("id1")))(Right(PostSubscriptionDetailsSuccessResponse))
+        val result = await(
+
+          TestRemoveBusiness.deleteBusiness(testReference, testBusinessId, Seq(testBusiness("id"), testBusiness("id1")))
+        )
+
+        result shouldBe Unit
+
+
```
[the remark this answers is not in the document]
Comment (AlexRimmerHMRC) [reply test/services/RemoveBusinessServiceSpec.scala:69]: If deleteSubscriptionDetails was called in this test, an exception would be thrown, again, happy to discuss

File: test/services/RemoveBusinessServiceSpec.scala (hunk 1 above)
Comment (AlexRimmerHMRC) [reply test/services/RemoveBusinessServiceSpec.scala:58]: 👍

**Review by sb-dev — DISMISSED** [review]

**Review by sb-dev — APPROVED** [review]

**Pull request merged**